In [1]:
import numpy as np
from scipy.stats import norm

---
## 📌 Question 1

A factory claims that the mean diameter of its bolts is **10 mm**.

A sample of **50 bolts** is taken, and the following statistics are observed:

- Sample mean (**x̄**) = **10.3 mm**
- Population standard deviation (**σ**) = **1.0 mm**
- Significance level (**α**) = **0.05**

Perform a **two-tailed one-sample Z-test** to evaluate the factory’s claim.

- Compute the **Z-statistic**
- Compare it with the **critical value**
- Clearly state your **conclusion**
- Also compute the **95% confidence interval** for the mean
---

#### Step 1 : Hypothesis
- H₀: μ = 10
- H₁: μ ≠ 10 (two-tailed test)

####  Step 2: Approach

- Population σ is known → use Z-test  
- Two-tailed test → check both directions  
- Compute:
  - Test statistic (z)
  - Critical value
  - p-value

In [2]:
# Given values
x_bar = 10.3
mu = 10
sigma = 1
n = 50
alpha = 0.05

#### Step 3: Compute Test Statistic

Formula: z = (x̄ - μ₀) / (σ / √n)

In [3]:
z = (x_bar - mu) / (sigma/ n**0.5)
z

2.1213203435596477

#### Step 4: Critical Value Method

For α = 0.05 (two-tailed):

α/2 = 0.025  
Left area = 1 - 0.025 = 0.975  

We need:
z such that P(Z < z) = 0.975

In [4]:
# Critical z-value
z_critical = norm.ppf(1-alpha/2)
z_critical

np.float64(1.959963984540054)

##### Decision
- |z| > z_critical 
- Reject H₀

#### Step 5: p-value Method

The p-value is computed as:

p = 2 × P(Z > |z|)

We use the survival function (sf), which directly gives the right-tail probability.

In [5]:
# p-value (two-tailed)
p = 2 * norm.sf(abs(z))
p

np.float64(0.033894853524688816)

##### Decision
- p < alpha
- Reject H₀

#### Step 6: Confidence Interval

CI = x̄ ± z_critical × (σ / √n)

In [6]:
margin = z_critical * (sigma / np.sqrt(n))
lower = x_bar - margin
upper = x_bar + margin

(lower, upper)

(np.float64(10.022819235130065), np.float64(10.577180764869937))

#### Final Conclusion

- Test statistic: z ≈ 2.12
- Critical value: ±1.96
- p-value ≈ 0.034

Since:
- |z| > z_critical
- p < α
- 10 is outside the confidence interval

We reject H₀.

The data provides sufficient evidence that the true mean diameter differs from 10 mm.

---
## 📌 Question 2

A survey of **800 voters** finds that **480 support** a new policy.

A pollster claims that **55% of the population** supports the policy.

- Sample size (**n**) = **800**
- Number of successes (**x**) = **480**
- Hypothesized proportion (**p₀**) = **0.55**
- Significance level (**α**) = **0.05**

Perform a **two-tailed one-sample proportion Z-test** to evaluate the claim.

- Compute the **Z-statistic**
- Compare it with the **critical value**
- Clearly state your **conclusion**
- Also compute the **95% confidence interval** for the proportion
---

##### Step 1: Hypothesis

- H₀: p = 0.55  
- H₁: p ≠ 0.55  (two-tailed test)

##### Step 2: Approach

- Data is in form of counts → use proportion test
- Compare observed proportion with claimed proportion
- Use Z-test for proportions

In [7]:
x = 480
n= 800
p0= 0.55
alpha2 = 0.05

In [8]:
p_hat = x/n
p_hat # Sample proportion

0.6

##### Step 3: Compute Z- Statics

In [9]:
# z = Observed value - Expected Value / Standard Error
z =  (p_hat - p0)/ (p0 *(1-p0)/n)**0.5
z

2.8426762180748018

##### Step 4: Critical Value Method
- Two Tailed Test, Alpha = 0.05

In [10]:
Z = norm.ppf(1-alpha/2)
Z

np.float64(1.959963984540054)

##### Decision Rule:
2.84 > 1.96 ~ 
|z|>Z:
- Reject H₀

##### Step 5: p-value method

In [13]:
p_val2= 2* norm.sf(abs(z))
p_val2

np.float64(0.004473649256611822)

##### Decision Rule:
0.0044 < 0.05 ~ 
p_value < alpha:
- Reject H₀

##### Step 6: Confidence Interval
CI=Estimate±(Critical Value×Standard Error)

In [12]:
Margin_of_error = Z * (p_hat* (1-p_hat)/ n)**0.5
lower = p_hat - Margin_of_error
upper = p_hat + Margin_of_error
lower,upper

(np.float64(0.5660524279777148), np.float64(0.6339475720222851))